homeassistant/components/matter/climate.py

```diff
@@ -0,0 +1,216 @@
+            return value / TEMPERATURE_SCALING_FACTOR
+        return None
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        return self._get_temperature_in_degrees(
+            clusters.Thermostat.Attributes.LocalTemperature
+        )
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach."""
+        match self.hvac_mode:
+            case HVACMode.COOL:
+                return self._get_temperature_in_degrees(
+                    clusters.Thermostat.Attributes.OccupiedCoolingSetpoint
+                )
+            case HVACMode.HEAT:
+                return self._get_temperature_in_degrees(
+                    clusters.Thermostat.Attributes.OccupiedHeatingSetpoint
+                )
+            case HVACMode.AUTO:
+                # When the system mode is set to "auto," there is no target temperature; instead, there is a target temperature low and high.
+                return None
+            case _:
+                return None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        target_temp: float | None = kwargs.get(ATTR_TEMPERATURE)
+
+        if target_temp is None or self.target_temperature is None:
+            return
+
+        temp_diff = int((target_temp - self.target_temperature) * 10)
+
+        match self.hvac_mode:
+            case HVACMode.HEAT:
+                command = clusters.Thermostat.Commands.SetpointRaiseLower(
+                    clusters.Thermostat.Enums.SetpointAdjustMode.kHeat,
+                    temp_diff,
+                )
+            case HVACMode.COOL:
+                command = clusters.Thermostat.Commands.SetpointRaiseLower(
+                    clusters.Thermostat.Enums.SetpointAdjustMode.kCool,
+                    temp_diff,
+                )
+            case HVACMode.AUTO:
+                # not sure how to handle this when system is in auto mode
+                command = clusters.Thermostat.Commands.SetpointRaiseLower(
+                    clusters.Thermostat.Enums.SetpointAdjustMode.kBoth,
+                    temp_diff,
+                )
+            case _:
+                return
+
+        await self.matter_client.send_device_command(
+            node_id=self._endpoint.node.node_id,
+            endpoint_id=self._endpoint.endpoint_id,
+            command=command,
+        )
+
+    @callback
+    def _update_from_device(self) -> None:
+        """Update from device."""
+        # work out supported features and modes
+        self.features = self.get_matter_attribute_value(
+            clusters.Thermostat.Attributes.FeatureMap
+        )
+
+
+# Discovery schema(s) to map Matter Attributes to HA entities
+DISCOVERY_SCHEMAS = [
+    MatterDiscoverySchema(
+        platform=Platform.CLIMATE,
+        entity_description=ClimateEntityDescription(key="MatterThermostat"),
```

Comment:
We want the climate entity to be the named after the device as the main entity.
```suggestion
        entity_description=ClimateEntityDescription(
            key="MatterThermostat",
            name=None,
        ),
```